pkg/registry/thirdpartyresourcedata/codec.go

```diff
@@ -0,0 +1,175 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package thirdpartyresourcedata
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/meta"
+	"k8s.io/kubernetes/pkg/expapi"
+	"k8s.io/kubernetes/pkg/runtime"
+)
+
+type thirdPartyResourceDataMapper struct {
+	mapper meta.RESTMapper
+	kind   string
+}
+
+func (t *thirdPartyResourceDataMapper) GroupForResource(resource string) (string, error) {
+	return t.mapper.GroupForResource(resource)
+}
+
+func (t *thirdPartyResourceDataMapper) RESTMapping(kind string, versions ...string) (*meta.RESTMapping, error) {
+	mapping, err := t.mapper.RESTMapping(kind, versions...)
```

Comment:
You are right. `kind` here is `ThirdPartyResourceData`, not a third-party kind. I misunderstood at first.